mautic/mautic
# Fix: Filter segments by select-multiple custom field

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/3717 https://github.com/mautic/mautic/issues/5530
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

Bugfix donate by www.newcom4u.ch

[//]: # ( Required: )
#### Description:
Impossible to create a segment based on a value on a customer field select-multiple
No contacts appears.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a custom field of select-multiple type. Use 2 (or more) different options, but setting the "label" different from "value".
2. Create a segment to filter one of these 2 types. The segment filter type is "including", and the value is one of the ones from the custom-field created above.
3. Run the cronjob to update segments. No contact get updated.
4. Update manually the lead table, setting the custom-field value to the "label", instead of the "value".
5. Run again the cronjob to update segments. This contact is added to the segment.

#### Steps to test this PR:
1.  Repeat all steps (don't forget re-save your segment)
2.  See if contacts was added properly to segment
